mobile/mysterium/openvpn_connection_setup.go

```diff
@@ -85,8 +87,45 @@ func (adapter channelToCallbacksAdapter) OnStats(openvpnStats openvpn3.Statistic
 // Openvpn3TunnelSetup is alias for openvpn3 tunnel setup interface exposed to Android/iOS interop
 type Openvpn3TunnelSetup openvpn3.TunnelSetup
 
-// OverrideOpenvpnConnection replaces default openvpn connection factory with mobile related one
-func (mobNode *MobileNode) OverrideOpenvpnConnection(tunnelSetup Openvpn3TunnelSetup) {
+// ReconnectableSession interface exposing reconnect for an active session
+type ReconnectableSession interface {
+	Reconnect(afterSeconds int) error
+}
+
+type sessionTracker struct {
+	session *openvpn3.Session
+	mux     sync.Mutex
+}
+
+func (st *sessionTracker) sessionCreated(s *openvpn3.Session) {
+	st.mux.Lock()
+	st.session = s
+	st.mux.Unlock()
+}
+
+// Reconnect reconnects active session after given time
+func (st *sessionTracker) Reconnect(afterSeconds int) error {
+	if st.session == nil {
```

Comment:
Should we protect the reading of `st.session` if we are doing it in other places?